### ISSUE TITLE
Create annotation that limits disabling extensions

### ISSUE THREAD

**Issue by nipafx** [junit-pioneer/junit-pioneer#477]
#366 will introduce an annotation `@DisabledUntil` that disables a test until a certain date. It was discussed whether other disabling annotations should get an attribute that has the same behavior. I don't think that's a good idea. Currently it's only planned to disable a test _until a date_, but _until a project version is reached_ or _until an issue is fixed_ are conceivable as well (and maybe others). Extending all disabling annotations with all these possibilities will bloat their APIs.

Instead I propose to create a new annotation `@Until` that gathers up this functionality and that all other disabling extensions check for, e.g.:

```java
@DisableIfDisplayName(contains = "st")
@Until(date = "2021-05-04")
void test() { /*...*/ }
```

This is how that could work:

* create an internal class for each check (date, version, closed issue, etc.) ~> test these thoroughly
* create an internal class that acts as a front and simply takes all attributes and forwards them (just for usability)
* carve out `@DisabledUntil` and forward to the usability class
* create a new annotation `@Until` that does nothing (except maybe check whether another disabling annotation is present and warn/fail if not)
* enhance all disabling extensions (where this makes sense) to check for the presence of `@Until` and, if it's found, forward the attributes to the usability class

If we consider requiring all disabling extension to mark their best-of date, then this approach has the downside that we can't have the compiler enforce setting the date by making `until` a required attribute in these other annotations. Then again, as soon as we add a second way to limit disabling (e.g. issue being resolved), then compiler can't help anymore either.

Long game: Advertise `@Until` in the wider community and offer a well-documented and maintained way for other projects to interact with it.

**Comment by nipafx** [comment]
I'm gonna mark this as _in discussion_ until I get some feedback on the idea and as _blocked_ until #366 is implemented.

**Comment by Michael1993** [comment]
> enhance all disabling extensions (where this makes sense) to check for the presence of @Until and, if it's found, forward the attributes to the usability class

I disagree. Instead it should check for the presence of a meta-annotation that is common among all "additional logic on disabling"-types. E.g.:
```java
@Retention(RUNTIME)
@Target(TYPE)
@DisablingLogic(UntilExtension.class)
public @interface Until {
  // properties, etc...
}
```
where `UntilExtension` implements
```java
public interface DisablingLogicCallback<A> {

  void consume(A annotation);

  boolean isDisablingAllowed();

}
```
I.e.: `class UntilExtension implements DisablingLogicCallback<Until>`

This way any additional logic can be seamlessly implemented without the need to further modify the behaviour of the disabling extensions themselves. *All class/method/field names are placeholders*

TL;DR: We should define an extension point as an interface and rely on that instead of checking specifically for `@Until`.
